File: programmers/lv1/모의고사.swift
//  프로그래머스 모의고사
//  21.02.10

import Foundation

func solution(_ answers:[Int]) -> [Int] {
  var ans: [Int] = []
  let a: [Int] = [1, 2, 3, 4, 5]
  let b: [Int] = [2, 1, 2, 3, 2, 4, 2, 5]
  let c: [Int] = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
  var score = Array(repeating: 0, count: 3)
  
  for i in 0..<answers.count {
    if answers[i] == a[i % a.count] { score[0] += 1 }
    if answers[i] == b[i % b.count] { score[1] += 1 }
    if answers[i] == c[i % c.count] { score[2] += 1 }
  }
  

  for i in 0...2 {
    if score[i] == score.max() {
      ans.append(i + 1)
    }
  }
  
  return ans
}

print(solution([1, 2, 3, 4, 5]))

File: programmers/lv1/모의고사.cpp
// 프로그래머스 모의고사
// 21.02.10
#include <iostream>
#include <algorithm>
#include <queue>
#include <vector>
#include <string>
#include <cmath>
#define INF 2147000000

typedef long long ll;

using namespace std;

int a[5] = {1,2,3,4,5};
int b[8] = {2,1,2,3,2,4,2,5};
int c[10] = {3,3,1,1,2,2,4,4,5,5};

vector<int> solution(vector<int> answers) 
{
    vector <int> ans;
    vector <int> score(3, 0);
    int len = answers.size();

    for (int i = 0; i < len; ++i)
    {
        if (a[i%5] == answers[i]) ++score[0];
        if (b[i%8] == answers[i]) ++score[1];
        if (c[i%10] == answers[i]) ++score[2];
    }

    int maxScore = *max_element(score.begin(), score.end());
    
    for (int i = 0; i < 3; ++i)
    {
        if (score[i] == maxScore) 
        {
            ans.push_back(i+1);
        }
    }
    
    return ans;
}